### ISSUE TITLE
AWOOOGA - website down ???

### ISSUE THREAD

**Issue by StuMackenzie** [MotusDev/Motus-TO-DO#1031]

**Comment by leberrigan** [comment]
Website is down

**Comment by cjardine-bsc** [comment]
I killed the lock. 

bah. 

Texting me is also an option in case I'm not on email.

**Comment by denislepage** [comment]
The NAS drive, which contains most of the data for sgdata, etc. is down. Kwic has ordered a replacement for the unit. They're hoping it will be on site tomorrow.

This will impact most of the sgdata process. I'll defer most of the procs queue items into that rely on the NAS.

**Comment by denislepage** [comment]
To be clear, this means that no new tag data will come in until this is resolved, Lotek or CTT.

**Comment by StuMackenzie** [comment]
user notice, or just let it ride for now? I'd suggest we wait to see if it's going to cause problems > 48 hrs out. We may want to consider a notice re. data backlogs

**Comment by leberrigan** [comment]
It seems like this has broken data uploads? We need to communicate this to collaborators network wide next time -- we're getting several emails about this.

**Comment by jsayers-bsc** [comment]
> 
> 
> It seems like this has broken data uploads? We need to communicate this to collaborators network wide next time -- we're getting several emails about this.

Agreed. Aside from the immediate NAS issue, there is the matter of the other processing backlog (which has already been inquired about by some users) and which the NAS failure has exacerbated. So there will be a lot more people wondering why their upload from last week hasn't showed up, why their connected receivers aren't up to date, and why they can't upload data. When it rains it pours!

**Comment by denislepage** [comment]
"Kwic" update. The unit is out of warranty, so we need to order a new one.  Costs should be around 2k, perhaps a bit more to increase the buffering capacity, which is likely worth it.

In the meantime, Andrew is looking at swapping the disks with our backup unit, so we may be able to restore service this afternoon. We'll have to revert the disk swap once the new unit is here. We probably need to anticipate a week for that to arrive, but it shouldn't matter too much (we'll just need to shut down processing for the swap).

We should discuss whether a failsafe unit would be worth the costs. For another NAS and disks, we'll probably looking at around 10k, but I'll work on some costs and we can discuss it.

**Comment by tve** [comment]
Any thought about moving to AWS or other quality cloud?

**Comment by denislepage** [comment]
We have, but the latency would be too high for some of the operations we’re doing now when they involve files of hundreds of GB, so it’d require some significant changes to our processing, so all current operations are done on local disks, and cloud is used for longer term storage. If we ever are able to get away from having to do periodic reprocessing, that approach might be more feasible. AWS is horribly slow to transfer large numbers of files. A single processing job may involve tens of thousands of small files, involving terabytes of data.

I’m looking into a High Availability system that relies on a failover cluster as the most viable alternative for now: https://www.synology.com/en-ca/dsm/feature/high_availability. Basically, 2 identical NAS each using RAID, sitting side-by-side, with one active, and one ready to take over in case of problems.

I welcome any other thoughts.

**Comment by tve** [comment]
Yeah, you can't just move storage and pull files over the internet for every operation. You'd want to move processing to AWS too. From what I see, it would solve a slew of problems.

FYI: <soapbox>From '99 to '06 I managed hosting of hundreds of servers in 4 datacenters across the globe, including internet routers to implement seamless site failover, etc. In '06 I switched to cloud exclusively (primarily AWS), running hundreds of servers and offering a product to automate large cloud infrastructure. I would never want to go back (unless there are very specific reasons, e.g. political or strategic). The benefits and savings are just so overwhelming.</soapbox>

**Comment by denislepage** [comment]
The backup NAS is now back online. Let's see if we can get things restarted now.

**Comment by denislepage** [comment]
Moving to cloud is an aspiration, but there are some roadblocks, particularly with the SGDATA vm. Recurring costs of cloud solutions are higher. Equipment costs are obviously the biggest factor of DIY hosting, but we have had better luck with finding funds for one-time purchase costs so far. Our current hosting arrangements are also somewhat favorable, too.

60-80 TB of storage with lots of I/O and data throughout is likely significantly higher than our current costs, but I may be wrong. I'd be keen to have some more in-depth chats and tap into your expertise on this topic.

Partnering with a ComputeCanada host is another option that we have considered, but this is limiting in other ways.

**Comment by tve** [comment]
Hey, setting my soapbox aside :-) this may not be an opportune time to look at cloud solutions plus I don't even know what I don't know about your operations and challenges :-) :-) . I was just struck by the fact that the two main issues that hit you this week are classic examples for why operating in the cloud is advantageous. Of course, everything is a trade-off and it's not like everything is paradise and perfect in AWS either, plus there's a definite learning curve.

WRT operating in the cloud being more expensive I really doubt that if you look at the big picture. At the highest level what you get by operating in the cloud is two things: flexibility and manpower savings. There's a whole set of issues you don't spend your human capital on anymore.

For example, disk fails: not your problem, you don't even really notice. On the flexibility side there are so many opportunities that may be difficult to even imagine right now. Just to take two examples. You have a big backlog: how about spinning up a few more boxes to process that? Can't do it in parallel due to software limitations? How about spinning up the fastest box available for a couple of days. On the other side, when things calm down and there's not much data to process: downsize to a smaller box to save money for those flood events.

Or you may want to change your schema as discussed a couple of days ago to center around runs instead of individual events (IIRC). How do you test that at scale? Well, you spin up a whole duplicate of your servers, do the transformation, test the site, hopefully declare victory and move production over. Unforeseen problem is discovered an hour after going live with the new site? (Naaah, never happens :-) You still have the old site running: move prod back and go find the issue. Once all sorted out, shut down and stop paying for the duplicate boxes. (Similar argument applies when a developer needs a big sandbox to mess with for a few days.)

When you're comparing costs you also have to take into account the whole life-cycle and what you really get. Yes, a 1TB disk is cheaper to buy than if you store that data for umpteen years in S3. But in S3 you pay for multi-site redundancy. And if you want to spin up 10 workers to process a backlog you will see zero performance degradation. And you spend zero ops person costs for all this.

I know the cap-ex vs. op-ex costs are an issue. The cloud providers do have some offerings that depending on your accounting can work as cap-ex. For example, many companies can use up-front committed payments to work as cap-ex. This is not my specialty, but I know it's a very, very common issue and an AWS rep would have some suggestions.

Something else, completely off the wall: what if collaborators could spin up their own boxes in the cloud "next" to yours and have high bandwidth access to the data so they can do their big data analysis right there in a standardised environment (as opposed to everyone downloading data and running it on whatever box with whatever software they may have)?

Overall, I think it all comes down to where you want to be and how you can get there given your resources :-). Anyway, apologies if I stuck my nose into something I shouldn't have. After all, I still have a good chunk of work on sensorgnome :-)

**Comment by denislepage** [comment]
No problem, I absolutely welcome that discussion.

Most of our environments are set as VM’s, so migrating them is probably not the key problem. We do have however a couple major bottlenecks. SGdata is probably the main issue on that front. It’s a bit of a Gordian knot. Once we get to the point where we might want to rewrite tagfinder/sgdata, I definitely think we should consider cloud based as one of the options. It may be difficult to do so until then.

Cost-wise, I’d be interested in working towards an estimate, even a rough one. If you are interested in helping, we could take this discussion offline, and I can send you some numbers to try and come up with a rough number. We’d pretty much need to move everything together, including many non Motus activities. We should have a good handle on our internal costs for comparison. Our situation is perhaps a bit unique, but once the capital costs are excluded, I’d be surprised if cloud comes on top. That said, there are definitely some other benefits and fewer potential headaches.

**Comment by tve** [comment]
> Cost-wise, I’d be interested in working towards an estimate, even a rough one. If you are interested in helping, we could take this discussion offline, and I can send you some numbers to try and come up with a rough number.

Sure! BTW, I don't know why "the capital costs are excluded" makes sense ;-)

What is SGdata?
Oh, looking for SGdata I noticed that there is https://github.com/MotusWTS/find_tags that has diverged from https://github.com/sensorgnome-org/find_tags: which is the real find_tags? :-)

**Comment by denislepage** [comment]
sgdata is the server where find_tag and motusServer are running and processing tag signals data.

I am surprised that find_tags is in sensorgnome-org. It doesn’t really belong there. We’ve been maintaining things related to sgdata in motuswts.

**Comment by Hussell** [comment]
It's alive!!

**Comment by denislepage** [comment]
Wonderful. We’ll need to plan for a bit of down time once the new unit arrives, but I’ll give a heads up.

**Comment by Hussell** [comment]
The process queues in the main databases need to be restarted.

Also, it looks like the connected receivers aren't connecting. I'm not sure what's going on there, yet.

**Comment by denislepage** [comment]
Good catch. I just restarted the import queues.

**Comment by StuMackenzie** [comment]
Someone mind popping a quarter in the machine?

**Comment by jsayers-bsc** [comment]
Is everything down again? It was all fine this morning but neither the website nor R package seem to be responding now...

**Comment by Hussell** [comment]
Website's working, for me, now.

**Comment by cjardine-bsc** [comment]
I killed the offending process.

**Comment by leberrigan** [comment]
So... I just uploaded data without thinking and it was processed within a couple hours. Does that mean we're all good, or do you want me to hold off before I invite the hoard of uploads?

**Comment by denislepage** [comment]
I suspect something changed in the database to increase the amount of locking that we’re getting.

We should try to hunt down the offending source, which isn’t necessarily going to be obvious, and may not be related to the processes that killed.

Looking at time involved in the procs queue would be a first step. Trying to think back about what might have changed in the past month would also help.

Stopping the imports of sandbox det data and sg data would likely help us too. I’ve started a new thread on another option (#1047), which relates to the duplication of data in sandbox.  It may not help with this lock issue, but it may help us in several ways, as these are relatively heavy operations.

**Comment by cjardine-bsc** [comment]
I'm on the same trail. 

One substantial change I made recently was the mods to the batches activity creation. This is only deployed in Sandbox, but it runs quite a bit more slowly than the old version (it does joins to GPS, and summarizes for all hourbins). 

https://github.com/MotusDev/Motus-TO-DO/issues/1016

I've not spent much time optimizing this. But it might be related to the locks we're seeing.

**Comment by denislepage** [comment]
We’re potentially caught up, but I’ll let Jeremy confirm.

Probably a good idea to let everyone know again, unless we need a couple extra days.

**Comment by Hussell** [comment]
The senorgnome.org website is still down, and therefore "connected" receivers aren't connecting. I can reach https://beta.sensorgnome.org/, for what that's worth.

**Comment by denislepage** [comment]
I'll see soon if I can log in. I used to

**Comment by Hussell** [comment]
Progress update: it turns out this is a network issue. The sensorgnome.org website is accessible from outside Birds Canada, so both servers are up, they just can't connect to each other.

**Comment by dfloer** [comment]
beta.sensorgnome.org is hosted on Linode, so that should generally be fine.
I took a poke at sensorgnome.org, the server/VM got rebooted, and something isn't right there. I'm having a hard time getting it started again the way I've started it before, I'll take another poke at it later (after some sleep), unless someone wants to look at it in the meantime.
This was one of the things that I was working on improving with beta., but that unfortunately didn't get finished before.

**Comment by dfloer** [comment]
> 60-80 TB of storage with lots of I/O and data throughout is likely significantly higher than our current costs, but I may be wrong. I'd be keen to have some more in-depth chats and tap into your expertise on this topic.

This sounds like something I can help with as well, considering I'm sitting in front of my 3-node Ceph cluster with about that amount of usable storage in it.

**Comment by StuMackenzie** [comment]
Thanks @dfloer - can you take a look at this and let us know what you think. 'connected' gnomes are not connecting and Jeremy is having trouble logging in from the office.

> 
> 
> beta.sensorgnome.org is hosted on Linode, so that should generally be fine. I took a poke at sensorgnome.org, the server/VM got rebooted, and something isn't right there. I'm having a hard time getting it started again the way I've started it before, I'll take another poke at it later (after some sleep), unless someone wants to look at it in the meantime. This was one of the things that I was working on improving with beta., but that unfortunately didn't get finished before.

**Comment by Hussell** [comment]
I'm getting 502 Bad Gateway now when I try to view https://sensorgnome.org - from home as well as from the office.

**Comment by StuMackenzie** [comment]
If we don't have a resolution by the end of Monday we should make an announcement. Would be nice to know that the issue is beforehand.

**Comment by leberrigan** [comment]
Looks like it's working now. Are you still troubleshooting or are we good to announce that it's fixed?

**Comment by StuMackenzie** [comment]
Let’s confirm that all Sensorgnome that should be connecting are connecting. I know some in Latin America aren’t and it would be worth checking on old cut and BSC HQ and any others you’re aware of. The data there was manually uploaded yesterday. 

____________________

**Comment by dfloer** [comment]
I've been poking at it for a couple hours, and I was just going to ask if it looked like it was working again. Good to hear that things are working again.

I'd say that it's "fixed" for now, but there may be some outages. I want to at least get it into a minimally maintainable state, which may require some (hopefully) short outages.

**Comment by StuMackenzie** [comment]
Deployments not connecting: all Costa rica

7970
5739
8279
8291
8214

____________________

**Comment by leberrigan** [comment]
BSC HQ is also missing, along with one other (Soley Cove; SG-1B0BRPI30C8D). It looks like it's only showing SG's which have connected in the past couple hours.

Some stats on the current list of stations showing on sensorgnome.org.

Latest Sync | Disconnected | Connected |  | Grand Total
-- | -- | -- | -- | --
05-Oct | 1 |   |   | 1
15-Oct |   | 1 |   | 1
21-Oct |   | 1 |   | 1
28-Oct |   | 1 |   | 1
31-Oct |   | 1 |   | 1
03-Nov |   | 1 |   | 1
04-Nov | 1 | 2 |   | 3
05-Nov | 4 | 38 |   | 42
15-Nov |   | 35 |   | 35
 |   |  |   | 
Grand   Total | 6 | 80 |   | 86

**Comment by Hussell** [comment]
Thank you, @dfloer. The processing of the backlog of data for the 77 receivers which reconnected was completed this morning.

**Comment by dfloer** [comment]
@StuMackenzie Have they connected yet, or are there still receivers that haven't connected that should be? From my end, it doesn't _look_ like there's anything else that's wrong.

**Comment by StuMackenzie** [comment]
The list above ‘should’ be connected. BSC HQ is now a sensrsration and looks like OLd Cut is checking in although the label is wrong. There are a few other inquiries out so we’ll have to see what feedback comes back from folks. But, see Thorsten’s notes re. ‘Test’ receivers.

____________________

**Comment by dfloer** [comment]
I still didn't figure out why Old Cut refuses to change in the status page, I'm pretty sure that's coming from the SQLite database, and that should have been changed a while ago.

**Comment by denislepage** [comment]
I am about to restart the server for maintenance, so Kwic can add the new drives. Motus will be down for a short while soon.

**Comment by leberrigan** [comment]
I can't connect to the management or and `/data/` pages right now.

**Comment by leberrigan** [comment]
It's up again

**Comment by j-sayers** [comment]
![image](https://user-images.githubusercontent.com/55001144/161064922-e6b549dc-507c-4616-908f-6c144a367cce.png)

Down again now. (We could probably use a better method of alerting and dealing with this than simply re-opening the old issue! Anyway for now....)

**Comment by leberrigan** [comment]
Thanks Josh. This is also affecting data downloads through R

**Comment by leberrigan** [comment]
Is there a way to make an alerting system for these outages? Most of the time we don't notice until somebody else tells us

**Comment by leberrigan** [comment]
Website is up again as of yesterday. The alerting system is already in place - not sure if it's possible to speed things up much more.

**Comment by j-sayers** [comment]
Website down, or at least unresponsive for the past half hour or so. No error messages

**Comment by j-sayers** [comment]
![image](https://user-images.githubusercontent.com/55001144/197089739-d838d08d-e661-4ca0-b549-cd2e545781d5.png)

![image](https://user-images.githubusercontent.com/55001144/197089822-2954fea2-9c4f-4683-9494-a3de5b92f32c.png)

Not sure I have anything more to add than the screenshots can say...

**Comment by denislepage** [comment]
I just restarted.

Tomcat ran out of memory, so there may be a leak somewhere.
